lib/scripts/update_agent_config_restart.sh

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+# Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2.0 License.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2022-Present Datadog, Inc.
+
+# This script is called by the node agent to expose CAPI metadata and DCA tags to the container agents
+# It sets the DD_NODE_AGENT_TAGS environment variable with these new tags
+# see: https://github.com/DataDog/datadog-agent/blob/7.40.x/pkg/cloudfoundry/containertagger/container_tagger.go#L131
+
+
+DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/.datadog}"
+SUPPRESS_DD_AGENT_OUTPUT="${SUPPRESS_DD_AGENT_OUTPUT:-true}"
+
+# correct way to export / source the .datadog_env file so that every variable is parsed
+export $(grep -v '^#' $DATADOG_DIR/.datadog_env | xargs)
+export DD_TAGS=$(LEGACY_TAGS_FORMAT=true python $DATADOG_DIR/scripts/get_tags.py node-agent-tags)
+
+# the agent cloud_foundry_container workloadmeta collector reads from this file
+# See: https://github.com/DataDog/datadog-agent/blob/main/pkg/workloadmeta/collectors/internal/cloudfoundry/cf_container/cloudfoundry_container.go#L24
+echo "$DD_TAGS" | awk '{ printf "%s", $0 }' > "$DATADOG_DIR/node_agent_tags.txt"
+
+# for debugging purposes
+printenv > /home/vcap/app/.datadog/.sourced_datadog_env
+
+# import helper functions
+source "$DATADOG_DIR/scripts/utils.sh"
+
+stop_datadog() {
+
+  # first try to stop the agent so we don't lose data and then force it
+  if [ -f "$DATADOG_DIR/run/agent.pid" ] || [ "$DD_LOGS_ENABLED" = "true" ] || [ "$DD_ENABLE_CHECKS" = "true" ]; then
```

Comment:
Let's do this for the other as well and instead of checking env vars or pid files, we check for the command's pid.
```suggestion
  if [ "$(find_pid ./agent)" = "" ]; then
```